src/gui/torrentcreatordialog.cpp

```diff
@@ -248,7 +248,16 @@ void TorrentCreatorDialog::handleCreationSuccess(const Path &path, const Path &b
         }
         params.useAutoTMM = false;  // otherwise if it is on by default, it will overwrite `savePath` to the default save path
 
-        BitTorrent::Session::instance()->addTorrent(result.value(), params);
+        if (BitTorrent::Session::instance()->isRestored())
+        {
+            BitTorrent::Session::instance()->addTorrent(result.value(), params);
+        }
+        else
+        {
+            QMessageBox::warning(this, tr("BitTorrent Session isn't restored")
+                                 , tr("BitTorrent Session isn't restored yet. Adding new torrents is impossible. Please wait...")
```

Comment:
This is quite bad usability, maybe just disable `m_ui->checkStartSeeding` checkbox in the constructor if Session isn't ready and add a tooltip about why it isn't available.

Then there is no need for this message.

Reply:
Done.